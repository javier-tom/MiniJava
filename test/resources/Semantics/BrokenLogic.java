class Main {
    public static void main(String[] args) {
        System.out.println((new Vars().run(10, 15)));
    }
}
class Vars {
    int a;
    int b;
    int c;
    boolean f1;
    boolean f2;

    public int run(int x, int y) { // x = 10, y = 15
        if (true < 1) { // true
            f1 = this.whiles(a, y);
        } else {
            if (false) {

            } else {

            }
        }
        return 1;
    }

    public boolean whiles(int a, int b) { // a = 0?, b = 15
        int[] arr;
        while (arr < b) {
            a = a + 1;
        }
        while (true){

        }
        while(1) {

        }
        arr = this.run3(a, b); // a = 15, b = 15
        return true;
    }

    public int[] run3(int a, int b) {
        int[] arr;
        arr = new int[a];
        a = a - b; // a = 0
        while(a < b - 10) { // ?
            arr[a] = 1;
            a = a + 1;
        }
        return arr;
    }
}