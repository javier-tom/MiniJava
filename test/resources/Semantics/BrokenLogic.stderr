Error on line 14: expected type int. Got type boolean
Error on line 28: expected type int. Got type int[]
Error on line 34: expected type boolean. Got type int